torch/csrc/jit/codegen/cuda/scheduler.h

```diff
@@ -13,20 +14,68 @@ TORCH_CUDA_API bool scheduleFusion(
     Fusion* fusion,
     const at::ArrayRef<c10::IValue> inputs);
 
+/**
+ * This struct communicates whether a launch param
+ * should be calculated per kernel instance if a new set of elements
+ * matches the kernel.
+ *
+ * @param value_ : -1 = symbolic, > 0 = valid constant launch param
+ * @param symbolic_value_ : a pointer to a symbolic val of param
+ */
+struct LaunchParam {
+  bool mutable_ = false;
+  int value_ = 1;
+
+  bool operator==(const LaunchParam& other) const {
+    // If the params are mutable, we don't care about the value
+    if (mutable_ && other.mutable_) {
```

Comment:
just a suggestion: it could be simplified as `return (mutable_ && other.mutable_) || value_ == other.value_;`